cusub_cortex/state_machine/src/tasks/start_gate_det.py

```diff
@@ -0,0 +1,685 @@
+                    if self.count:
+                        self.drive_client.increment_setpoint(0.2, loop=False)
+                    else:
+                        self.drive_client.increment_setpoint(0.6, loop=False)
+                else:
+                    # Not within threshold
+                    self.count = 0
+                    self.drive_client.increment_setpoint(-0.2, loop=False)
+
+                self.listener.clear_new_dv_flag(dobj_num)
+
+            if userdata.timeout_obj.timed_out:
+                self.yaw_client.disable()
+                self.drive_client.disable()
+                self.depth_client.disable()
+                userdata.outcome = "timed_out"
+                return "not_found"
+
+            # Detected 2nd Pole
+            if len(dobj_nums) > 1:
+                self.cuprint("Detected 2nd Pole")
+                break
+                
+            self.depth_client.set_setpoint(-1.5, loop=False)
+            r.sleep()
+
+        self.yaw_client.disable()
+        self.drive_client.disable()
+        self.depth_client.disable()
+        return "found_2nd_pole"
+
+
+class Approach_2_Poles(Objective):
+    outcomes = ['found_3rd_pole', 'timed_out', 'lost_object']
+
+    target_class_id = "start_gate_pole"
+
+    def __init__(self, task_name, listener):
+        name = task_name + "/Approach_2_Poles"
+        super(Approach_2_Poles, self).__init__(self.outcomes, name)
+        self.listener = listener
+        self.yaw_client = PIDClient(name, "yaw")
+        self.drive_client = PIDClient(name, "drive") #, "cusub_common/motor_controllers/mag_pid/")
+        self.depth_client = PIDClient(name, "depth")
+
+        seconds = rospy.get_param("tasks/start_gate/seconds_in_postion", 2)
+        self.rate = rospy.get_param("tasks/start_gate/new_dv_check_rate", 30)
+        self.count_target = seconds * self.rate
+        self.count = 0
+
+        self.mag_target = rospy.get_param("tasks/start_gate/mag_target")
+        self.elev_target = rospy.get_param("tasks/start_gate/elev_target")
+        self.elev_thresh = rospy.get_param("tasks/start_gate/elev_thresh")
+
+
+    def execute(self, userdata):
+        self.cuprint("executing")
+        self.toggle_waypoint_control(True)
+
+        # Find start_gate pole dobject number and check for errors
+        dobj_nums = self.listener.query_class(self.target_class_id)
+        if len(dobj_nums) > 1: # Check if more than 1 instance of target class
+            self.cuprint(str(len(dobj_nums)) + " " + self.target_class_id + " classes detected!", warn=True)
+            self.cuprint("selecting the second", warn=True)
+        elif not dobj_nums: # Check if target class is not present (shouldn't be possible)
+            self.cuprint("somehow no " + self.target_class_id + " classes found?", warn=True)
+            return "lost_object"
+        # DOBJECT NUMBER
+        self.cuprint(str(dobj_nums), warn=True)
+        dobj_num = dobj_nums[1]
```

Comment:
yeah this code gets repeated in all of our tasks, so we should make it something like an inherited method